frontend/src/components/TradeBox/EncryptedChat/index.tsx

```diff
@@ -1,522 +1,56 @@
-import React, { useEffect, useState } from 'react';
-import { useTranslation } from 'react-i18next';
-import {
-  Button,
-  IconButton,
-  Tooltip,
-  TextField,
-  Grid,
-  Container,
-  Card,
-  CardHeader,
-  Paper,
-  Typography,
-} from '@mui/material';
-import { encryptMessage, decryptMessage } from '../../../pgp';
-import { saveAsJson } from '../../../utils';
-import { AuditPGPDialog } from '../../Dialogs';
-import RobotAvatar from '../../RobotAvatar';
-import { systemClient } from '../../../services/System';
-import { websocketClient, WebsocketConnection } from '../../../services/Websocket';
-
-// Icons
-import CheckIcon from '@mui/icons-material/Check';
-import CloseIcon from '@mui/icons-material/Close';
-import ContentCopy from '@mui/icons-material/ContentCopy';
-import VisibilityIcon from '@mui/icons-material/Visibility';
-import CircularProgress from '@mui/material/CircularProgress';
-import KeyIcon from '@mui/icons-material/Key';
-import { ExportIcon } from '../../Icons';
-import { useTheme } from '@mui/system';
+import React, { useState } from 'react';
+import EncryptedSocketChat from './EncryptedSocketChat';
+import EncryptedTrutleChat from './EncryptedTrutleChat';
 
 interface Props {
+  turtleMode: boolean;
   orderId: number;
+  takerNick: string;
+  makerNick: string;
   userNick: string;
+  chatOffset: number;
 }
-
-interface EncryptedChatMessage {
+export interface EncryptedChatMessage {
```

Comment:
@Reckless-Satoshi I see on your PR you have other models, maybe we need to sync on this

Reply:
Will take care of it when that PR needs to be merged. This one will go first.